### PR TITLE
[6.0] Deleting CMS Filesystem Package

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This PR removes the CMS Filesystem package. You can use the Joomla framework Filesystem package as a drop-in replacement.

This PR depends on several other PRs to remove the cores dependencies on these classes:

- [ ] #43359: File::makesafe()
- [ ] #43958: Folder to framework
- [x] #43362: Rest of File to framework

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_joomlaupdate/finalisation.php [hunk 1]
```diff
@@ -237,11 +237,11 @@ public static function delete(string $folderName): bool
         }
     }
 
-    if (!class_exists('\Joomla\CMS\Filesystem\File')) {
+    if (!class_exists('\Joomla\Filesystem\File')) {
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/finalisation.php:240]: This seems wrong to me as you create 1 line below an alias to that not existing class.

File: administrator/components/com_joomlaupdate/finalisation.php [hunk 2]
```diff
@@ -237,11 +237,11 @@ public static function delete(string $folderName): bool
...
         class_alias('\\Joomla\\Filesystem\\File', '\\Joomla\\CMS\\Filesystem\\File');
     }
 
-    if (!class_exists('\Joomla\CMS\Filesystem\Folder')) {
+    if (!class_exists('\Joomla\Filesystem\Folder')) {
```
Comment (richard67) [inline administrator/components/com_joomlaupdate/finalisation.php:244]: Same as above.

File: administrator/components/com_joomlaupdate/finalisation.php (hunk 1 above)
Comment (HLeithner) [reply administrator/components/com_joomlaupdate/finalisation.php:240]: makes no sense to me too, the full if statements doesn't make sense to me

Comment (richard67) [reply administrator/components/com_joomlaupdate/finalisation.php:240]: Yes, those 2 if statements in lines 240 to 246 don't make sense like they have been changed by this PR (alias to the not existing class), and without this PR they only made sense for a scenario where an update is made and for some reason the old code where the CMS stuff is still used is still running.

Now when the CMS stuff will be removed I think these 2 if statements can be removed, as we require 5.4 for updating to 6, and in 5.4 there should be nothing left in the update process which uses the CMS filesystem classes.

**Comment by Hackwar** [comment]
I'm going to close this PR the moment I have created a new one which moved the package to the compat plugin.

**Pull request closed without merging**